packages/core/src/components/StepList/StepList.jsx

```diff
@@ -0,0 +1,80 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Step from './Step';
+
+/**
+ * The `StepList` component is the preferred method for building a step list,
+ * since it handles all the state logic necessary to produce its markup. A
+ * step is represented by an object with text, progress, and routing information
+ * and can optionally include an array of substeps as well as a description.
+ */
+export const StepList = ({ steps, ...props }) => (
+  <ol className="ds-c-step-list ds-u-margin-top--4">
+    {steps.map(step => <Step step={step} key={step.id} {...props} />)}
+  </ol>
+);
+
+// Define the shape of a single step so we can recursively define the shape
+// as well as reuse it in multiple components' prop-type definitions. This
+// has to be in this file or else the docs generator will break.
+export const stepShape = {
+  id: PropTypes.string.isRequired,
```

Comment:
I think I left a comment on this before, but just flagging I'm still not sure why an `id` is a required prop

Reply:
Because it's used as the key when listing steps

Reply:
Does this need to be explicitly defined or can we generate it? Seems tedious to require it if we can generate it. We use `lodash.uniqueid` in a few other components to generate ID's for a11y purposes. Could we do that for these?

Reply:
I mean, really I could have it use the array index. That would be really simple, and the order of these should not be changing. I was just avoiding it because I already had it as a required field because they were necessary for our app. I'll have it use the id if it's there and the index if it's not

Reply:
I just didn't want to make my map-function code any larger